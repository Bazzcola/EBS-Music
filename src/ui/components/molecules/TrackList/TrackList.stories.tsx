import React from 'react';
import { TrackList } from './TrackList';
import { MemoryRouter } from 'react-router-dom';

export default {
  title: 'Molecules/TrackList',
  component: TrackList
};

export const Track = () => {
  return (
    <MemoryRouter>
      <TrackList />
    </MemoryRouter>
  );
};
