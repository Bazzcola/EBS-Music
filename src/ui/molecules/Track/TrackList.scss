.track_list {
  max-width: 900px;
  margin: 0 auto;

  .track_list__title {
    display: flex;
    justify-content: space-between;
    height: 50px;
    border-top: 1px grey solid;
    border-bottom: 1px grey solid;
    padding: 0px 43px 0px 35px;

    h3 {
      display: flex;
      align-items: center;
    }
  }

  .track_item {
    display: flex;
    justify-content: space-between;
    padding: 0px 30px 0px 81px;
    color: #ef5466;
  }
}

.track_text {
  margin-right: 700px;
  margin-left: 30px;
}
