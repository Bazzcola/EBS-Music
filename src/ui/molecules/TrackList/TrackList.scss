.container {
  width: 100%;
  padding: 0 15px;
}

.track_list {
  max-width: 900px;
  margin: 0 auto;

  .track_list__title {
    display: flex;
    height: 50px;
    border-top: 1px grey solid;
    border-bottom: 1px grey solid;
    padding: 0px 43px 0px 35px;
    margin-top: 20px;
    margin-bottom: 10px;

    h3 {
      display: flex;
      align-items: center;
    }
  }

  .track_item {
    display: flex;
    padding: 0px 30px 0px 20px;
    color: #ef5466;
    align-items: center;
  }
}

.track_text {
  margin-left: 30px;
}

.track_text__item {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.date_item {
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin-left: 20px;
}