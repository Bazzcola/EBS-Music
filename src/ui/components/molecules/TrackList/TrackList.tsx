import React, { useContext } from 'react';
import { CircleButton } from 'ui/components/atoms/CircleButton/CircleButton';
import { Button } from 'ui/components/atoms/Button/Button';
import { Tracks } from 'ui/context/audioContext';
import { Link } from 'react-router-dom';
import { AudioContext } from 'ui/context/audioContext';
import './TrackList.scss';

export const TrackList = () => {
  const typeButton = 'pink';
  const typeCircleButton = 'black';
  const classType = 'circle_button';
  const { currentSong } = useContext(AudioContext);
  const { audioFiles } = useContext(AudioContext);

  return (
    <div className="music_list_container">
      <div className="container">
        <div className="track_list">
          <Button typeButton={typeButton} />
          <div className="track_list__title">
            <h3>#</h3>
            <div className="track_text__item">
              <p className="track_text">TRACK</p>
              <p>L.</p>
            </div>
          </div>
          {Tracks.map(
            (item: {
              id: number;
              title: string;
              duration: string;
              src: string;
            }) => (
              <div
                className={
                  currentSong && item.src === audioFiles
                    ? 'track_item_active'
                    : 'track_item'
                }
                key={item.src}
              >
                <CircleButton
                  classType={classType}
                  type={typeCircleButton}
                  src={item.src}
                  id={item.id}
                />
                <div className="date_item">
                  <Link to={`/${item.title}`}>
                    <p>{item.title}</p>
                  </Link>
                  <p>{item.duration}</p>
                </div>
              </div>
            )
          )}
        </div>
      </div>
    </div>
  );
};
