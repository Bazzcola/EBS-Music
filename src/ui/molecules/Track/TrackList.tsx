import React from 'react';
import { CircleButton } from 'ui/atoms/CircleButton/CircleButton';
import { Button } from 'ui/atoms/Button/Button';
import './TrackList.scss';

export const TrackList = () => {
  const item = [
    {
      name: 'Alabama',
      time: '3:20'
    },
    {
      name: 'Alabama',
      time: '3:20'
    },
    {
      name: 'Alabama',
      time: '3:20'
    },
    {
      name: 'Alabama',
      time: '3:20'
    }
  ];
  return (
    <div className="container">
      <Button />
      <div className="track_list">
        <div className="track_list__title">
          <h3>#</h3>
          <p className="track_text">TRACK</p>
          <p>L.</p>
        </div>
        {item ? (
          item.map((item: { name: string; time: string }) => (
            <div className="track_item" key={Math.random()}>
              <CircleButton />
              <p>{item.name}</p>
              <p>{item.time}</p>
            </div>
          ))
        ) : (
          <div className="track_item">
            <p>Name</p>
            <p>Time</p>
          </div>
        )}
      </div>
    </div>
  );
};
