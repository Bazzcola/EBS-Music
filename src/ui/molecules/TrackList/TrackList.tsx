import React from 'react';
import { CircleButton } from 'ui/atoms/CircleButton/CircleButton';
import { Button } from 'ui/atoms/Button/Button';
import { Tracks } from 'ui/context/audioContext';
import './TrackList.scss';


export const TrackList = () => {
  const typeButton = 'pink';
  const typeCircleButton = 'black';
  const classType = 'circle_button';
 
  return (
    <div className="container">
      <div className="track_list">
        <Button typeButton={typeButton} />
        <div className="track_list__title">
          <h3>#</h3>
          <div className="track_text__item">
            <p className="track_text">TRACK</p>
            <p>L.</p>
          </div>
        </div>
        {
          Tracks.map((item: {id:number; title:string; duration:string}) => (
            <div className="track_item" key={item.id}>
              <CircleButton classType={classType} type={typeCircleButton} />
              <div className="date_item">
                <p>{item.title}</p>
                <p>{item.duration}</p>
              </div>
            </div>
          ))
        }
      </div>
    </div>
  );
};
