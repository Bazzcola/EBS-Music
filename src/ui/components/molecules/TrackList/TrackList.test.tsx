import React from 'react';
import { render } from '@testing-library/react';
import { TrackList } from './TrackList';
import { MemoryRouter } from 'react-router-dom';

it('renders TrackList correctly', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <TrackList />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
