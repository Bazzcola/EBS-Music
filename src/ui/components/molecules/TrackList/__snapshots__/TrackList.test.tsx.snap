// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TrackList correctly 1`] = `
<DocumentFragment>
  <div
    class="music_list_container"
  >
    <div
      class="container"
    >
      <div
        class="track_list"
      >
        <button
          class="big_button big_button_undefined"
        >
          <svg
            class="play_svg"
          >
            <path
              d="M6 11A5 5 0 1 0 6 1a5 5 0 0 0 0 10zm0 1A6 6 0 1 1 6 0a6 6 0 0 1 0 12zM4.937 4.017a.125.125 0 0 0-.187.108v3.75c0 .096.104.156.187.108l3.25-1.875a.125.125 0 0 0 0-.216l-3.25-1.875z"
            />
          </svg>
          Play
        </button>
        <div
          class="track_list__title"
        >
          <h3>
            #
          </h3>
          <div
            class="track_text__item"
          >
            <p
              class="track_text"
            >
              TRACK
            </p>
            <p>
              L.
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track1"
            >
              <p>
                Track1
              </p>
            </a>
            <p>
              6:13
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track2"
            >
              <p>
                Track2
              </p>
            </a>
            <p>
              7:06
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track3"
            >
              <p>
                Track3
              </p>
            </a>
            <p>
              5:44
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track4"
            >
              <p>
                Track4
              </p>
            </a>
            <p>
              5:03
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track5"
            >
              <p>
                Track5
              </p>
            </a>
            <p>
              5:54
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track6"
            >
              <p>
                Track6
              </p>
            </a>
            <p>
              4:40
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track7"
            >
              <p>
                Track7
              </p>
            </a>
            <p>
              7:01
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track8"
            >
              <p>
                Track8
              </p>
            </a>
            <p>
              5:25
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track9"
            >
              <p>
                Track9
              </p>
            </a>
            <p>
              6:29
            </p>
          </div>
        </div>
        <div
          class="track_item"
        >
          <button
            class="circle_button circle_button_black"
          >
            <svg
              class="svg_stop svg_stop_circle_button"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
          <div
            class="date_item"
          >
            <a
              href="/Track10"
            >
              <p>
                Track10
              </p>
            </a>
            <p>
              8:47
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
